src/modules/core-functions.js

```diff
@@ -0,0 +1,199 @@
+import { getData, sendData } from './api-utils.js';
+import FoodList from './food-list.js';
+
+const foodList = new FoodList();
+
+const FOOD_API_BASE_URL = 'https://www.themealdb.com/api/json/v1/1/';
+const ALL_FOOD_ENDPOINT = 'filter.php?a=Jamaican';
+const ALL_FOOD_API_URL = FOOD_API_BASE_URL + ALL_FOOD_ENDPOINT;
+
+const INV_API_BASE = 'https://us-central1-involvement-api.cloudfunctions.net/'
++ 'capstoneApi/apps/';
+const INV_API_KEY = 'zX9lc5HNiZeTfJrwouGw';
+const LIKES_ENDPOINT = '/likes';
+const COMMENT_ENDPOINT = '/comments';
+
+const foodListWrapper = document.getElementById('home');
+const commentPopup = document.getElementById('comment-popup');
+const itemCounter = document.getElementById('counter');
+
+export function getComments(id) {
+  return new Promise((resolve) => {
+    const parameter = `?item_id=${id}`;
+    const COMMENT_API = INV_API_BASE + INV_API_KEY + COMMENT_ENDPOINT + parameter;
+    getData(COMMENT_API).then((commentsFromAPI) => {
+      if (commentsFromAPI.error) commentsFromAPI = [];
+      foodList.addComments(id, commentsFromAPI);
+      resolve();
+    });
+  });
+}
+
+function postComment(id, input, textarea) {
+  const ADD_COMMENT_URL = INV_API_BASE + INV_API_KEY + COMMENT_ENDPOINT;
+  const data = {
+    item_id: id,
+    username: input.value,
+    comment: textarea.value,
+  };
+
+  sendData(ADD_COMMENT_URL, data).then((res) => {
+    if (res.status === 201) {
+      input.value = '';
+      textarea.value = '';
+      const commentWrapper = document.getElementById('comments');
+      commentWrapper.innerHTML += `<li class="comment">
+        <h4 class="comment-author">${data.username}</h4>
+        <p class="comment-message">${data.comment}</p>
+        </li> `;
+    }
+  });
+}
+
+export function displayPopUp(id) {
+  commentPopup.classList.add('show');
+  commentPopup.innerHTML = `      <div id="close-popup-button">&times;</div>
+    <div class="comment-popup-wrapper">
+      <img id="food-image" src="${foodList.foods[id].image}" alt="" />
+      <h2 id="food-title">${foodList.foods[id].title}</h2>
+      <p id="food-description">...</p>
+
+      <h3 id="comments-header">Comments:</h3>
+      <ul id="comments">
+      </ul>
+
+      <h3>Add Comment:</h3>
+      <form id="comment-form">
+        <input type="text" name="name" id="nameField" placeholder="Your Name" required />
+        <textarea
+          name="comment"
+          rows="5"
+          placeholder="Your Comment"
+          required
+          id="messageField"
+        ></textarea>
+        <button id="button">Submit</button>
+      </form>
+    </div>`;
+
```

Comment:
- [ ] Please, consider also fixing catching the empty comment adding. You can add a warning message maybe, if the inputs are empty and submit button will not actually submit.
 > Here I have entered an empty comment and I am able to do it:
![2022-03-11 21 39 52](https://user-images.githubusercontent.com/49064106/157924909-4f8102fc-f4ec-45eb-95fd-febd94185dda.gif)

Reply:
Thanks for noticing this issue. it is fixed now.